tests/test_consistency_checker.py

```diff
@@ -568,3 +572,36 @@ def test_networks_consistency():
             },
         ),
     )
+
+
+def test_cloudspec_consistency():
+    cloud_spec = CloudSpec(
+        type="lxd",
+        endpoint="https://127.0.0.1:8443",
+        credential=CloudCredential(
+            auth_type=CloudAuthType.client_certificate_auth_type,
+            attributes={
+                "client-cert": "foo",
+                "client-key": "bar",
+                "server-cert": "baz",
+            },
+        ),
+    )
+
+    assert_consistent(
+        State(cloud_spec=cloud_spec, model=Model(name="lxd-model", type="lxd")),
+        Event("start"),
+        _CharmSpec(
+            MyCharm,
+            meta={"name": "MyVMCharm"},
+        ),
+    )
+
+    assert_inconsistent(
+        State(cloud_spec=cloud_spec, model=Model(name="k8s-model", type="kubernetes")),
+        Event("start"),
+        _CharmSpec(
+            MyCharm,
+            meta={"name": "MyVMCharm"},
```

Comment:
Probably a different name makes more sense.

```suggestion
            meta={"name": "MyK8sCharm"},
```

Reply:
that, or 'Luke' works too